src/components/Gradebook/StatusAlerts.test.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { StatusAlerts, mapDispatchToProps, mapStateToProps } from './StatusAlerts';
+import { closeBanner } from '../../data/actions/grades';
+
+describe('StatusAlerts', () => {
+  const renderStatusAlerts = (params) => {
+    const alertParams = {
+      handleCloseSuccessBanner: jest.fn().mockName('handleCloseSuccessBanner'),
+      showSuccessBanner: false,
+      ...params,
+    };
+    const alerts = renderer.create(<StatusAlerts {...alertParams} />);
+    expect(alerts.toJSON()).toMatchSnapshot();
+  };
+
+  describe('error alerts', () => {
+    it('not rendered when filters are valid', () => {
+      renderStatusAlerts({
+        isMinCourseGradeFilterValid: true,
+        isMaxCourseGradeFilterValid: true,
+      });
+    });
+
+    it.each([
+      [false, true],
+      [true, false],
+      [false, false],
+    ])('rendered when course grade filters are invalid.', (isMinCourseGradeFilterValid, isMaxCourseGradeFilterValid) => {
+      renderStatusAlerts({
+        isMinCourseGradeFilterValid,
+        isMaxCourseGradeFilterValid,
+      });
+    });
+  });
+
+  describe('success alert', () => {
+    it('rendered when param is true', () => {
+      renderStatusAlerts({
+        showSuccessBanner: true,
+      });
+    });
+
+    it('not rendered otherwise', () => {
+      renderStatusAlerts({
+        showSuccessBanner: false,
+      });
+    });
+  });
+
+  describe('mapStateToProps', () => {
+    it('showSuccessBanner', () => {
+      expect(mapStateToProps({ grades: { showSuccess: true } }).showSuccessBanner).toBeTruthy();
```

Comment:
This is a place where I would just set a weird value and track across.
Alternatively, I have set up in the past a set of libraries for checking selectors that are a bit more brute-force like this, but wrapped in libraries.